GeoLib/SurfaceGrid.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * \date 2012-09-22
+ * \brief Definition of the SurfaceGrid class.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ */
+
+#include <algorithm>
+#include <bitset>
+
+#include <logog/include/logog.hpp>
+
+#include "SurfaceGrid.h"
+
+#include "Surface.h"
+#include "Triangle.h"
+
+#include "MathLib/Point3d.h"
+
+namespace GeoLib {
+
+SurfaceGrid::SurfaceGrid(Surface const*const sfc) :
+	AABB(sfc->getAABB()), _n_steps({{1,1,1}})
+{
+	// enlarge the bounding, such that the points with maximal coordinates
+	// fits into the grid
+	for (std::size_t k(0); k<3; ++k) {
+		_max_pnt[k] += std::abs(_max_pnt[k]) * 1e-6;
+		if (std::abs(_max_pnt[k]) < std::numeric_limits<double>::epsilon()) {
+			_max_pnt[k] = (_max_pnt[k] - _min_pnt[k]) * (1.0 + 1e-6);
+		}
+	}
+
+	std::array<double, 3> delta{{ _max_pnt[0] - _min_pnt[0],
+		_max_pnt[1] - _min_pnt[1], _max_pnt[2] - _min_pnt[2] }};
+
+	if (delta[0] < std::numeric_limits<double>::epsilon()) {
```

Comment:
Since std::numeric_limits is called many times in the entire source code, I suggest to set several constants in a namespace for that, e.g.
```
namespace Constant
{
    const double double_epsion = std::numeric_limits<double>::epsilon();
    const double double_max = std::numeric_limits<double>::max();
   ...
}
```
This could save a little compilation time. If interest, I will may prepare a PR concerning to this point.